crates/ethrpc/src/block_stream/mod.rs

```diff
@@ -205,12 +237,17 @@ pub trait BlockRetrieving: Debug + Send + Sync + 'static {
 #[async_trait::async_trait]
 impl BlockRetrieving for Web3 {
     async fn current_block(&self) -> Result<BlockInfo> {
-        get_block_info_at_id(self, BlockNumber::Latest.into()).await
+        get_block_info_at_id(self, BlockNumber::Latest.into())
+            .await?
+            .try_into()
     }
 
     async fn block(&self, number: u64) -> Result<BlockNumberHash> {
         let block = get_block_info_at_id(self, U64::from(number).into()).await?;
-        Ok((block.number, block.hash))
+        Ok((
+            block.number.context("missing block_number")?.as_u64(),
```

Comment:
why is this change needed? :thinking:

Reply:
`get_block_info_at_id` no longer returns `BlockInfo` but a `Block<H256>` where these value can be `None`.
That being said I should rename the function. 😅